cdap-ui/app/cdap/components/PreviewData/index.tsx

```diff
@@ -147,22 +163,28 @@ const PreviewDataViewBase: React.FC<IPreviewDataViewProps> = ({
     </div>
   );
 
-  const showRecordView = tableData.inputFieldCount >= 100 || tableData.outputFieldCount >= 100;
-
   return (
     <div>
       <If condition={!previewId}>{noPreviewDataMsg(classes)}</If>
       <If condition={previewLoading}>{loadingMsg(classes)}</If>
       <If condition={error}>{errorMsg(classes)}</If>
 
-      <If condition={!previewLoading && previewId && !isEmpty(previewData) && !showRecordView}>
+      <If condition={!previewLoading && previewId && !isEmpty(previewData) && viewMode === 'Table'}>
```

Comment:
Maybe we can extract these logic in If condition into a function and add a comment!